hedera-node/hedera-admin-service-impl/src/main/java/com/hedera/node/app/service/admin/impl/handlers/UnzipUtility.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (C) 2020-2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.admin.impl.handlers;
+
+import java.io.BufferedOutputStream;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Utility class to unzip a zip file.
+ * <p>
+ * This class was copied from hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/utils/UnzipUtility.java
+ * Once we are wholly migrated to the modularized services, we can remove the other location.
+ * */
+public final class UnzipUtility {
+    private static final Logger log = LogManager.getLogger(UnzipUtility.class);
+
+    private static final int BUFFER_SIZE = 4096;
+
+    private UnzipUtility() {
+        throw new UnsupportedOperationException("Utility Class");
+    }
+
+    public static boolean unzip(final byte[] bytes, final String dstDir) throws IOException {
+        final File destDir = new File(dstDir);
+        if (!destDir.exists()) {
+            if (!destDir.mkdirs()) {
+                log.fatal("Unable to create the directory for update assets: {}", destDir);
+                return false;
+            }
+            log.info("Created directory {} for update assets", destDir);
+        }
+
+        final var zipIn = new ZipInputStream(new ByteArrayInputStream(bytes));
+        ZipEntry entry = zipIn.getNextEntry();
+        if (entry == null) return false;
+        while (entry != null) {
+            var filePath = dstDir + File.separator + entry.getName();
+            final var fileOrDir = new File(filePath);
+            filePath = fileOrDir.getCanonicalPath();
+
+            if (!filePath.startsWith(destDir.getCanonicalPath() + File.separator)) {
+                throw new IllegalArgumentException("Zip file entry " + filePath + " has an invalid path prefix!");
```

Comment:
this method returns null in some cases and throws an Exception in others. Can you make the behavior more clear